test/odbc/psqlodbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,1016 @@
+  
+
+  const string INSERT_STRING = "(" + std::to_string(PK_INSERTED) + "," + DATA_INSERTED + ")";
+  const string UPDATE_WHERE_CLAUSE = COL1_NAME + " = " + std::to_string(PK_INSERTED);
+
+  const int PK_BYTES_EXPECTED = 4;
+  const int DATA_BYTES_EXPECTED = 8;
+
+  int pk;
+  long long int data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const vector<tuple<int, int, SQLPOINTER, int, SQLLEN *>> BIND_COLUMNS = {
+    {1, SQL_C_LONG, &pk, 0, &pk_len},
+    {2, SQL_C_SBIGINT, &data, 0, &data_len}
+  };
+
+  vector<pair<string, string>> update_col = {
+    {COL2_NAME, DATA_UPDATED_VALUE}
+  };
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table using the correct ODBC data type mapping.
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, INSERT_STRING));
+  odbcHandler.CloseStmt();
+
+  // Bind Columns
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(BIND_COLUMNS));
+
+  // Assert that value is inserted properly
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string>{COL1_NAME}));
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(pk_len, PK_BYTES_EXPECTED);
+  ASSERT_EQ(pk, PK_INSERTED);
+  ASSERT_EQ(data_len, DATA_BYTES_EXPECTED);
+  ASSERT_EQ(data, StringToBigInt(DATA_INSERTED));
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+  odbcHandler.CloseStmt();
+
+  // Update value and assert an error is present
+  rcode = SQLExecDirect(odbcHandler.GetStatementHandle(), (SQLCHAR *)UpdateTableStatement(TABLE_NAME, update_col, UPDATE_WHERE_CLAUSE).c_str(), SQL_NTS);
+  ASSERT_EQ(rcode, SQL_ERROR);
+  odbcHandler.CloseStmt();
+
+  // Assert that no values changed
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string>{COL1_NAME}));
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(pk_len, PK_BYTES_EXPECTED);
+  ASSERT_EQ(pk, PK_INSERTED);
+  ASSERT_EQ(data_len, DATA_BYTES_EXPECTED);
+  ASSERT_EQ(data, StringToBigInt(DATA_INSERTED));
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_BigInt, Arithmetic_Operators) {
+  vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+  const int BYTES_EXPECTED = 8;
+
+  long long int pk;
+  long long int data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const vector<string> INSERTED_PK = {
+    "-73708",
+    "123",
+    "233"
+  };
+
+  const vector<string> INSERT_DATA = {
```

Comment:
nit: Let's be a bit more consistent with our naming and call this `INSERTED_DATA` since we have a variable named `INSERTED_PK`.